test/system/executable/tests.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2020 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+    Tests for the ddsrouter executable.
+
+    Contains a package of system test for ddsrouter tool
+
+    usage: test.py -e <binary_path> -c <config_file>
+
+    binary_path: DDS Router binary path
+
+"""
+
+import argparse
+import logging
+import os
+import signal
+import subprocess
+import sys
+import time
+
+DESCRIPTION = """Script to execute DDS Router executable test"""
+USAGE = ('python3 tests.py -e <path/to/ddsrouter-executable>'
+         ' [-t LIST[test-name]] [-d]')
+
+# Sleep time to let process init and finish
+SLEEP_TIME = 1
+MAX_SIGNALS_SEND_ITERATIONS = 3
+
+def parse_options():
+    """
+    Parse arguments.
+
+    :return: The arguments parsed.
+    """
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+        add_help=True,
+        description=(DESCRIPTION),
+        usage=(USAGE)
+    )
+    required_args = parser.add_argument_group('required arguments')
+    required_args.add_argument(
+        '-e',
+        '--exe',
+        type=str,
```

Comment:
In the type you can specify a `class`, a function or a `lambda` to validate the input. I'd check here if the directory exists. I'd check also that the config-file exists and have execution permissions

Reply:
:+1: